File: merge-sort/mergeSort.js
function mergeSort(arr) {
  let results = [];
  let i = 0;
  let j = 0;
  if (arr.length <= 1) return arr;
  let mid = Math.floor(arr.length / 2);
  let left = mergeSort(arr.slice(0, mid));
  let right = mergeSort(arr.slice(mid));
  while (i < left.length && j < right.length) {
    if (right[j] > left[i]) {
      results.push(left[i]);
      i++;
    } else {
      results.push(right[j]);
      j++;
    }
  }
  while (i < left.length) {
    results.push(left[i]);
    i++;
  }
  while (j < right.length) {
    results.push(right[j]);
    j++;
  }
  return results;
}

console.log(mergeSort([24, 26, 47, 2, 5, 6, 8, 9, 12, 15, 20]));

File: algorithms/merge_sort/merge_sort.py
def mergeSort(arr):
    if len(arr) <= 1:
        return arr
    results = []
    i = 0
    j = 0
    mid = len(arr) // 2
    left = mergeSort(arr[:mid])
    right = mergeSort(arr[mid:])
    while (i < len(left) and j < len(right)):
        if (right[j] > left[i]):
            results.append(left[i])
            i += 1
        else:
            results.append(right[j])
            j += 1
    while (i < len(left)):
        results.append(left[i])
        i += 1
    while (j < len(right)):
        results.append(right[j])
        j += 1
    return results


print(mergeSort([24, 26, 47, 2, 5, 6, 8, 9, 12, 15, 20]))
